### PR TITLE
promisify and lipstickify

### PR DESCRIPTION
**Pull request by SamVerschueren**
Did my best but I'm not sure about the return value of the method. Currently it returns a promise-like `kill` function. This means the result is a function that can be used to kill the pop-up, but can also be used as promise.

Should we actually support that? Can't we just return `execa` directly? It returns the child process as well so if people want to kill it, they still can.

### CODE REVIEW

**Comment by sindresorhus** [comment]
> Instead, we can pass the kill method as the first argument in the promise instead.

I don't really see any point in returning the kill method from the function. Do you? Instead, I think we should just supply the kill function as the first argument in the promise as explained above.

```js
websitePopup().then(close => {
      close();
});
```

**Comment by SamVerschueren** [comment]
But is it necessary to call the child process' kill method after the popup was closed by the user? Because the promise resolves the moment the user closes the popup. So if that's not necessary, why bother to return that function at all?

**Comment by sindresorhus** [comment]
Oh, lol, I thought it resolved when it opened.

File: index.js [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
-		if (err) {
-			return cb(err.killed ? null : err);
-		}
+	const cp = execa('automator', ['-i', opts.url, tempWrite.sync(wflow)]);
 
-		cb();
-	});
+	const kill = cp.kill.bind(cp);
+	kill.then = cp.then;
+	kill.catch = cp.catch;
```
Comment (sindresorhus) [inline index.js]: Don't you have to `bind` these to the `cp` instance?

File: index.js [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
+const tempWrite = require('temp-write');
 
-module.exports = function (opts, cb) {
-	if (typeof opts !== 'object' && typeof opts.url === 'string') {
-		throw new TypeError('`options.url` required');
-	}
+const tpl = fs.readFileSync(path.resolve(__dirname, 'popup.wflow'), 'utf8');
+
+module.exports = opts => {
+	opts = opts || {};
```
Comment (sindresorhus) [inline index.js]: Use `Object.assign()`, and you can set the default width/height here instead.

Comment (SamVerschueren) [inline index.js]: :+1:

File: index.js (hunk 1 above)
Comment (SamVerschueren) [inline index.js]: Not sure to be honest. They are bound to the promise at the [execa](https://github.com/sindresorhus/execa/blob/master/index.js#L181-L182) level. So if we bind again, it won't be bound to the promise anymore but to the childprocess, no?

**Comment by sindresorhus** [comment]
Looks good :) ✨

**Comment by SamVerschueren** [comment]
Awesome! :)

**Pull request merged**